dwave_networkx/algorithms/markov.py

```diff
@@ -0,0 +1,205 @@
+    >>> potentials = {('a', 'b'): {
+        (0, 0): -1,
+        (0, 1): .5,
+        (1, 0): .5,
+        (1, 1): 2
+        }}
+    >>> MN = dnx.markov_network(potentials)
+    >>> sampler = dimod.ExactSolver()
+    >>> samples = dnx.sample_markov_network(MN, sampler)
+    >>> samples[0]
+    {'a': 0, 'b': 0}
+
+    >>> import dimod
+    ...
+    >>> potentials = {('a', 'b'): {
+        (0, 0): -1,
+        (0, 1): .5,
+        (1, 0): .5,
+        (1, 1): 2
+        }}
+    >>> MN = dnx.markov_network(potentials)
+    >>> sampler = dimod.ExactSolver()
+    >>> samples = dnx.sample_markov_network(MN, sampler, return_sampleset=True)
+    >>> samples.first
+    Sample(sample={'a': 0, 'b': 0}, energy=-1.0, num_occurrences=1)
+
+    Notes
+    -----
+    Samplers by their nature may not return the optimal solution. This
+    function does not attempt to confirm the quality of the returned
+    sample.
+
+    """
+
+    # we could do the fixing with dimod's FixVariablesComposite but for
+    # for backwards compatibility we'll do it on construction
+    bqm = markov_network_bqm(MN, fixed=fixed)
+
+    sampleset = sampler.sample(bqm, **sampler_args)
+
+    if return_sampleset:
+        return sampleset
+    else:
+        return list(map(dict, sampleset.samples()))
+
+
+def markov_network_bqm(MN, fixed=None):
+    """Construct a binary quadratic model for a markov network.
+
+
+    Parameters
+    ----------
+    G : NetworkX graph
+        A Markov Network as returned by :func:`.markov_network`
+
+    fixed : dict
+        A dictionary of variable assignments to be fixed in the markov network.
+
+    Returns
+    -------
+    bqm : :obj:`dimod.BinaryQuadraticModel`
+        A binary quadratic model.
+
+    """
+
+    bqm = dimod.BinaryQuadraticModel.empty(dimod.BINARY)
+
+    # the variable potentials
+    for v, ddict in MN.nodes(data=True, default=None):
+        potential = ddict.get('potential', None)
+
+        if potential is None:
+            continue
+
+        # for single nodes we don't need to worry about order
+
+        phi0 = potential[(0,)]
+        phi1 = potential[(1,)]
+
+        bqm.add_variable(v, phi1 - phi0)
+        bqm.add_offset(phi0)
+
+    # the interaction potentials
+    for u, v, ddict in MN.edges(data=True, default=None):
+        potential = ddict.get('potential', None)
+
+        if potential is None:
+            continue
+
+        # in python<=3.5 the edge order might not be consistent so we use the
+        # one that was stored
+        order = ddict['order']
+        u, v = order
+
+        phi00 = potential[(0, 0)]
+        phi01 = potential[(0, 1)]
+        phi10 = potential[(1, 0)]
+        phi11 = potential[(1, 1)]
+
+        bqm.add_variable(u, phi10 - phi00)
+        bqm.add_variable(v, phi01 - phi00)
+        bqm.add_interaction(u, v, phi11 - phi10 - phi01 + phi00)
+        bqm.add_offset(phi00)
+
+    if fixed is not None:
+        # dimod >= 0.8.0 has a bqm.fix_variables method but we'll do them
```

Comment:
Would a try/catch be more efficient? Assuming `bqm.fix_variables` is optimized in some way...

Reply:
You would think but in this case it is literally the same...

Reply:
Well then, disregard. :)